corehq/apps/accounting/models.py

```diff
@@ -2451,6 +2494,31 @@ def get_credits_by_subscription_and_features(cls, subscription,
         ).all()
 
     @classmethod
+    def get_total_credits(cls, subscription=None, account=None, product_type=None, feature_type=None):
```

Comment:
It looks like this method does 1 of 4 things:
- gets credits for a subscription with a product
- gets credits for a subscription with a feature
- gets credits for an account with a product
- gets credits for an account with a feature
It also looks like you can pass both a subscription and an account into this method, but only the account credit_lines will get counted. 
I know this pattern happens elsewhere in the accounting code, but I find this pretty confusing. Might be worth breaking this into 4 (or 2) different methods that call this one: `get_total_credits_for_account`, `get_total_credits_for_subscription` or something like that.
Then, each of those methods will be easily unit testable, which is awesome.